django/contrib/admin/checks.py

```diff
@@ -331,11 +331,14 @@ def _check_fields(self, obj):
                     id="admin.E005",
                 )
             ]
-        fields = flatten(obj.fields)
-        if len(fields) != len(set(fields)):
+        field_counts = collections.Counter(flatten(obj.fields))
+        if duplicate_fields := sorted(
+            [field for field, count in field_counts.items() if count > 1]
+        ):
```

Comment:
You can do without the list comprehension here:
```suggestion
        if duplicate_fields := sorted(
            field for field, count in field_counts.items() if count > 1
        ):
```

Reply:
@g-nie yeah it was partly just for consistent ordering for the sake of the tests but sorting can happen only on errors to reduce possibly unneeded work, thanks for the suggestion